api/edge_api/identities/tasks.py

```diff
@@ -40,10 +41,14 @@ def call_environment_webhook_for_feature_state_change(
         return
 
     feature = Feature.objects.get(id=feature_id)
-    changed_by = FFAdminUser.objects.get(id=changed_by_user_id)
+    match changed_by_user_id:
+        case str():
+            changed_by = MasterAPIKey.objects.get(id=changed_by_user_id).name
+        case _:
+            changed_by = FFAdminUser.objects.get(id=changed_by_user_id).email
```

Comment:
This is too hacky for my taste.

Is there a reason we can't add a new `changed_by_user_type` task argument?

Reply:
I agree with @khvn26 on this.